### PR TITLE
THRIFT-4403: Fix incorrect usage of 'this'

### PR DESCRIPTION
**Pull request by bananer**

### CODE REVIEW

**Comment by jeking3** [comment]
Great - I am going to see if this resolves any of the cross test known failures.

File: lib/js/src/thrift.js [hunk 1]
```diff
@@ -599,8 +599,8 @@ Thrift.TWebSocketTransport.prototype = {
           //If the user made calls before the connection was fully
           //open, send them now
           this.send_pending.forEach(function(elem) {
-             this.socket.send(elem.buf);
-             this.callbacks.push((function() {
+             self.socket.send(elem.buf);
```
Comment (jeking3) [inline lib/js/src/thrift.js]: On line 597, self is set to be equal to this, so what does this change actually do?

Comment (bananer) [reply lib/js/src/thrift.js]: In short, `this` inside of a `function()` in JavaScript is not transferred reliably from outer scopes. Try setting a breakpoint at this line and you will see that `self` and `this` are different things here.

It's a bit of a race condition to have this block actually execute, but calling a service method right after opening the transport crashed here repeatedly in firefox.

Long version: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this

Comment (jeking3) [reply lib/js/src/thrift.js]: Perhaps line 597 should be removed then?

Comment (jeking3) [reply lib/js/src/thrift.js]: I'm running through cross tests to see if this resolves anything we know failed in the past.

Comment (jeking3) [reply lib/js/src/thrift.js]: We don't run "js" through cross test right now. :|

**Pull request closed without merging**